registries/deep_rag.hocon

```diff
@@ -0,0 +1,180 @@
+"""
+                        },
+                    }
+                },
+            },
+            "instructions": """
+Examine the names of the files in the file_list and take into account any user_description
+of the file naming structure to come up with a JSON structure
+which describes the following:
+* A short "name" for the body of work that is being pieced up which must be less that 40 characters.
+* A "description" of the overall strategy for how the files are to be grouped.
+    If you must mention the word 'files' in your description, use the word 'content' instead.
+* A "groups" list which contains one JSON entry for each of the groups, each entry of which has:
+    1. A short "name" for the group which is unique, less than 40 characters and
+        is the primary reference to the group in snake case.
+    2. A "description" of the group and how it is unique from the other groups.
+    3. A "files" dictionary whose keys are each of the files that belong in the group and
+       whose values are a snake case translation of the file name.
+
+Constraints:
+* Each and every file in the file_list must be represented in
+  at least one group.
+* Any single group may have at least one, but never more than 7 files in it.
+* There can be at most 7 groups.
+* Whenever possible follow the advice in the grouping_constraints, if that is present.
+
+Output the JSON structure described.
+""",
+        },
+        {
+            "name": "create_network",
+            "function": {
+                "description": "Processes the grouping_json structure to create a network.",
+                "parameters": {
+                    "type": "object",
+                    "properties": {
+                        "files_directory": {
+                            "type": "string"
+                        },
+                        "grouping_json": {
+                            "type": "object",
+                            "properties": {
+                                "name": {
+                                    "type": "string"
+                                },
+                                "description": {
+                                    "type": "string"
+                                },
+                                "groups": {
+                                    "type": "array",
+                                    "items": {
+                                        "type": "object"
+                                         # Not necessary to go into detail yet. Trusting the caller.
+                                    }
+                                }
+                            },
+                            "required": ["name", "description", "groups"]
+                        }
+                    },
+                    "required": ["files_directory", "grouping_json"]
+                }
+            },
+
+            "class": "create_networks.CreateNetworks",
```

Comment:
Separate coded tool to create networks based on the user input.
Hence all this "reservations": true business.